spec.md

```diff
@@ -179,6 +220,47 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+### Public Modules
+
+A public module is a module that is marked with the `public`{.firrtl} keyword.
+A public module may be instantiated outside by other circuits.
+
+The example below shows a public module `Foo`:
+
+``` firrtl
+public module Foo:
+  input a: UInt<1>
+  output b: UInt<1>
+
+  connect b, a
+```
+
+A public module has a number of restrictions:
+
+1. A public module may have no ports of uninferred width.
+1. A public module may have no ports of abstract reset type.
+1. A public module may have no ports of input probe type.
+1. A `RWProbe`{.firrtl} may not be used to access a public module's ports.
+1. A public module's ports are lowered using one of the port lowering ABIs
+   defined in the FIRRTL ABI specification.
```

Comment:
This doesn't seem like it should be a "restriction". This is a commentary on this proposal's relation to the ABI. 

We should scootch it down a few lines, instead of having it appear in this list. And I haven't read the latest version of the ABI spec, but when this says "one of the port lowering ABIs", the question that comes to my mind is, "which one? And who decides?"

Reply:
Also, remove "The last point requires clarification" from the next sentence. This is just evidence that commentary on ABI is a piece of commentary and not a restriction :)

Reply:
Agree. This isn't a restriction. I'll drop it. You're question about which one/who decides is important and the current spec is silent on this. The only way I know of to properly fix this is to add the port lowering convention to each external module and any public module. For now, the spec can choose to be silent about this. However, we should add this.

Will drop the "requires clarification" sentence.

Reply:
Fixed in 90f91c0.